cookbooks/env_vars/recipes/cloud.rb

```diff
@@ -8,21 +8,24 @@
   perform_restart = node['env_vars']['perform_restart']
 
   node['dna']['engineyard']['environment']['apps'].each do |app_data|
-    template "/data/#{app_data['name']}/shared/config/env.cloud" do
+    app_name = app_data['name']
+
+    template "/data/#{app_name}/shared/config/env.cloud" do
       source "env.cloud.erb"
       owner ssh_username
       group ssh_username
       mode 0744
       variables(:environment_variables => fetch_environment_variables(app_data))
       helpers(EnvVars::Helper)
 
-      notifies :run, "execute[restart_#{app_data['name']}]", :delayed
+      notifies :run, "execute[restart_#{app_name}]", :delayed
     end
 
-    execute "restart_#{app_data['name']}" do
-      command "if [ -d /data/#{app_data['name']}/current ]; then /engineyard/bin/app_#{app_data['name']} restart; fi"
+    execute "restart_#{app_name}" do
+      command "if [ -d /data/#{app_name}/current ]; then /engineyard/bin/app_#{app_name} restart; fi"
```

Comment:
I think it makes sense to move both conditions at single expression or even remove this condition from bash script (`if [ -d /data/#{app_name}/current ];`) so end up with:

```ruby
execute "restart_#{app_name}" do
  command "/engineyard/bin/app_#{app_name}"
  user ssh_username
  action :nothing
  only_if { perform_restart && ::File.exist?("/engineyard/bin/app_#{app_name}") && ::File.exist?("/data/#{app_data['name']}/current}")}
end
```

Reply:
Also I am not sure about `perform_restart`, this command won't be executed by default (`action :nothing`) and only run when file `env.cloud` is changed and notifies this script. So I think you should drop this condition.

Reply:
@rzaharenkov +1. I'll update the PR.